android/src/main/java/com/brentvatne/exoplayer/ExoPlayerView.kt

```diff
@@ -0,0 +1,289 @@
+import android.view.View
+import android.view.ViewGroup
+import android.widget.FrameLayout
+import androidx.core.content.ContextCompat
+import androidx.media3.common.AdViewProvider
+import androidx.media3.common.C
+import androidx.media3.common.Player
+import androidx.media3.common.Tracks
+import androidx.media3.common.VideoSize
+import androidx.media3.common.text.Cue
+import androidx.media3.common.util.Assertions
+import androidx.media3.common.util.UnstableApi
+import androidx.media3.exoplayer.ExoPlayer
+import androidx.media3.ui.SubtitleView
+import com.brentvatne.common.api.ResizeMode
+import com.brentvatne.common.api.SubtitleStyle
+import com.brentvatne.common.api.ViewType
+import com.brentvatne.common.toolbox.DebugLog
+
+@UnstableApi
+class ExoPlayerView(private val context: Context) :
+    FrameLayout(context, null, 0),
+    AdViewProvider {
+
+    private var surfaceView: View? = null
+    private var shutterView: View
+    private var subtitleLayout: SubtitleView
+    private var layout: AspectRatioFrameLayout
+    private var componentListener: ComponentListener
+    private var player: ExoPlayer? = null
+    private var layoutParams: ViewGroup.LayoutParams = ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT)
+    private var adOverlayFrameLayout: FrameLayout
+    val isPlaying: Boolean
+        get() = player != null && player?.isPlaying == true
+
+    @ViewType.ViewType
+    private var viewType = ViewType.VIEW_TYPE_SURFACE
+    private var hideShutterView = false
+
+    init {
+        componentListener = ComponentListener()
+
+        val aspectRatioParams = LayoutParams(
+            LayoutParams.MATCH_PARENT,
+            LayoutParams.MATCH_PARENT
+        )
+        aspectRatioParams.gravity = Gravity.CENTER
+        layout = AspectRatioFrameLayout(context)
+        layout.layoutParams = aspectRatioParams
+
+        shutterView = View(context)
+        shutterView.layoutParams = layoutParams
+        shutterView.setBackgroundColor(ContextCompat.getColor(context, android.R.color.black))
+
+        subtitleLayout = SubtitleView(context)
+        subtitleLayout.layoutParams = layoutParams
+        subtitleLayout.setUserDefaultStyle()
+        subtitleLayout.setUserDefaultTextSize()
+
+        updateSurfaceView(viewType)
+
+        adOverlayFrameLayout = FrameLayout(context)
+
+        layout.addView(shutterView, 1, layoutParams)
+        layout.addView(adOverlayFrameLayout, 2, layoutParams)
+
+        addViewInLayout(layout, 0, aspectRatioParams)
+        addViewInLayout(subtitleLayout, 1, layoutParams)
+    }
+
+    private fun clearVideoView() {
+        if (surfaceView is TextureView) {
+            player?.clearVideoTextureView(surfaceView as TextureView)
+        } else if (surfaceView is SurfaceView) {
+            player?.clearVideoSurfaceView(surfaceView as SurfaceView)
+        }
+    }
+
+    private fun setVideoView() {
+        if (surfaceView is TextureView) {
```

Comment:
```suggestion
when (surfaceView) {
        is TextureView -> player?.setVideoTextureView(surfaceView as TextureView)
        is SurfaceView -> player?.setVideoSurfaceView(surfaceView as SurfaceView)
        else -> {
            Log.w("setVideoView", "Unexpected surfaceView type: ${surfaceView?.javaClass?.name}")
        }
    }
```

Reply:
@l2hyunwoo @seyedmostafahasani 
Looks good!
But How about suggestions below?
```
        when (val view = surfaceView) {
            is TextureView -> player?.setVideoTextureView(view)

            is SurfaceView -> player?.setVideoSurfaceView(view)

            else -> {
                Log.w("setVideoView", "Unexpected surfaceView type: ${view?.javaClass?.name}")
            }
        }
```
I think we need to avoid type cast with `as` if possible.

Reply:
I agree with your suggestion.

Reply:
👍🏻 Thanks @YangJonghun I [applied](https://github.com/TheWidlarzGroup/react-native-video/pull/4038/commits/c2b1f243d84278f8a7700f8b13ae7b762d40cb1a) your suggestion!